sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -362,12 +362,26 @@ class Analyzer(
       exprs.flatMap {
         case s: Star => s.expand(child, resolver)
         case e =>
-          e.transformDown {
+          e.transformUp {
+            // ResolveFunctions can handle the case when the number of variables is not valid
             case f1: UnresolvedFunction if containsStar(f1.children) =>
               f1.copy(children = f1.children.flatMap {
                 case s: Star => s.expand(child, resolver)
                 case o => o :: Nil
               })
+            case c: CreateStruct if containsStar(c.children) =>
```

Comment:
Not sure if we have the other functions that can accept `star` as an input parameter. If so, I think we need to create a `trait` for all these case classes. Then, we can remove the duplicate code. Any better idea? Thanks! : )